react-web/src/components/SessionDetailsForm.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+
+import { UsernameForm } from './UsernameForm';
+
+const USERNAME_STORAGE_KEY = 'userName';
+
+class SessionDetailsForm extends React.Component {
+  
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sessionId: null,
+      [USERNAME_STORAGE_KEY]: null
+    };
+  }
+
+  componentDidMount = () => {
+    this.setState({
```

Comment:
Would it be possible to initialize these in the constructor instead? I think this would cause the component to render twice?

Reply:
:+1: